shell/browser/api/electron_api_web_contents.cc

```diff
@@ -1523,13 +1523,18 @@ void WebContents::HandleNewRenderFrame(
   // Set the background color of RenderWidgetHostView.
   auto* web_preferences = WebContentsPreferences::From(web_contents());
   if (web_preferences) {
-    absl::optional<SkColor> maybe_color = web_preferences->GetBackgroundColor();
-    web_contents()->SetPageBaseBackgroundColor(maybe_color);
-
+    auto maybe_color = web_preferences->GetBackgroundColor();
     bool guest = IsGuest() || type_ == Type::kBrowserView;
-    SkColor color =
+
+    // If webPreferences has no color stored we need to explicitly set guest
+    // webContents background color to transparent.
+    auto pb_bg_color =
+        guest ? maybe_color.value_or(SK_ColorTRANSPARENT) : maybe_color;
+    web_contents()->SetPageBaseBackgroundColor(pb_bg_color);
+
+    auto bg_color =
         maybe_color.value_or(guest ? SK_ColorTRANSPARENT : SK_ColorWHITE);
```

Comment:
Why not the following instead ?

```suggestion
       auto bg_color =
        maybe_color.value_or(guest ? SK_ColorTRANSPARENT : SK_ColorWHITE);
       web_contents()->SetPageBaseBackgroundColor(bg_color);
       SetBackgroundColor(rwhv, bg_color);
```

Reply:
I wrote it this way initially because i intentionally wanted to defer to the behavior of `SetPageBaseBackgroundColor` when `maybe_color` is invalid (as opposed to `SetBackgroundColor`, where we choose a color explicitly) - I'm open to changing it but feel it might be more futureproof to separate them.

Reply:
`SK_ColorWHITE` seems to be the default page base bg color https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/core/exported/web_view_impl.cc;l=3225. I think it is better to merge them to reflect the defaults, we can separate them if the default choices defer.

Reply:
Sounds good! i'll update that :)